### ISSUE TITLE
Fails to push metrics to Stackdriver: java.lang.IllegalArgumentException: fromIndex(0) > toIndex(-1)

### ISSUE THREAD

**Issue by straki** [micrometer-metrics/micrometer#2045]
Hi,

I'm using micrometer 1.4.2 with spring boot 2.2.6, Java 11 and Kotlin 1.3.61

This is Stackdriver configuration:

```
@Profile("!local")
@Configuration
class MicrometerConfig {
    @Bean
    fun timedAspect(meterRegistry: MeterRegistry?): TimedAspect {
        return TimedAspect(meterRegistry!!)
    }

    @Bean
    fun stackdriverConfig(@Value("\${spring.cloud.gcp.config.project-id}") projectId: String): StackdriverConfig {
        return object : StackdriverConfig {
            override fun get(key: String): String? {
                return null
            }

            override fun projectId(): String {
                return projectId
            }

            override fun batchSize(): Int {
                return 199
            }
        }
    }

    @Bean
    fun stackdriverMeterRegistry(stackdriverConfig: StackdriverConfig?): StackdriverMeterRegistry {
        return StackdriverMeterRegistry.builder(stackdriverConfig!!).build()
    }
}
```

I already had some issues before when bach size was bigger so that's why it's 199.

The thing is that service starts sending and send 2-3 times after that every other time is an error.

<img width="1497" alt="Screenshot 2020-04-29 at 17 51 15" src="https://user-images.githubusercontent.com/3831239/80617743-c4a76200-8a42-11ea-8dcc-12c34607c0b3.png">
<img width="1324" alt="Screenshot 2020-04-29 at 17 49 18" src="https://user-images.githubusercontent.com/3831239/80617759-c8d37f80-8a42-11ea-8036-10ca1b141439.png">

Is there any configuration for this or some advice what should I do.

Thanks in advance

**Comment by izeye** [comment]
@straki Sorry for the trouble.

It seems to be caused by 2718921956863c5fa6f83f186bd3a718fa910282.

I created https://github.com/micrometer-metrics/micrometer/pull/2047 to revert it.

**Comment by straki** [comment]
@izeye thanks for a quick reply.

Is there a version that I can rollback not to have this problem, what do you suggest?

**Comment by izeye** [comment]
@straki Based on 2718921956863c5fa6f83f186bd3a718fa910282, it was included in 1.4.2, so 1.4.1 should be okay.

**Comment by cristiammercado** [comment]
Hi, 

Any progress on this issue? I am using version 1.5.0. Should I also downgrade to 1.4.1?
